src/rustfmt/main.rs

```diff
@@ -43,7 +43,27 @@ fn main() {
 
 /// Format Rust code
 #[derive(Debug, StructOpt, Clone)]
-#[structopt(name = "rustfmt", version = include_str!(concat!(env!("OUT_DIR"),"/version-info.txt")))]
+#[structopt(
+    name = "rustfmt",
+    version = include_str!(concat!(env!("OUT_DIR"),"/version-info.txt")),
+    about = r#"Format Rust code
+
+Rustfmt runs on a set of files or stdin. When invoked without any file arguments, rustfmt will
+read code from stdin.
+
+Please visit https://rust-lang.github.io/rustfmt to see all rustfmt configuration options.
+
+EXAMPLES
+
+    cat lib.rs | rustfmt
+        Feed the contents of "lib.rs" to rustfmt via stdin
+
+    rustfmt --emit=files lib.rs main.rs
+        Run rustfmt over "lib.rs" and "main.rs", formatting in-place
+
+    rustfmt --config-path=rustfmt.toml --print-config=current
+        Print the resolved rustfmt configuration formed by rustfmt.toml
```

Comment:
I love the inclusion of examples here! I'm a little torn though on the options listed in the last two :thinking: 

On the one hand what's shown is the most commonly used, default values for emit mode and config path, but OTOH, those flags aren't necessary _because_ they're using the default values and I wouldn't want folks to to think they have to include `--emit=files`

Reply:
Ah thanks for the tip, I usually use from stdin so i didnt even realize the default is to files when they are passed :)

I updated example to use `emit=stdout`. I kept the print config one the way it is because you have to pass `--print-config` to get it to work anyway.